PatternPal/PatternPal.Core/Checks/ParameterCheck.cs

```diff
@@ -16,6 +28,104 @@ internal class ParameterCheck : CheckBase
         RecognizerContext ctx,
         INode node)
     {
-        throw new NotImplementedException();
+        IParameterized hasParameters = CheckHelper.ConvertNodeElseThrow<IParameterized>(node);
+
+        // Retrieve the parameters the method has to compare to list of TypeChecks and convert to
+        // IEntities.
+        List< IEntity > methodParameters =
+            hasParameters.GetParameters().Select(x => ctx.Graph.Relations.GetEntityByName(x)).ToList();
+
+        List<ICheckResult> subCheckResultsResults = new List<ICheckResult>();
+
+        // Method has no parameters
+        if (!methodParameters.Any())
+        {
+            return new NodeCheckResult
+            {
+                ChildrenCheckResults = subCheckResultsResults,
+                FeedbackMessage = $"The method has no parameters.",
+                Priority = Priority
+            };
+        }
+
+        // For each typecheck, check whether one of the parameters of method has the correct type
+        // and if so remove from method parameters list.
+        foreach (TypeCheck typecheck in _parameterTypes)
+        {
+            bool noneCorrect = true;
+
+            // There are no parameters to check with.   
+            if (methodParameters.Count == 0)
+            {
+                // TODO
+                ICheckResult temp = new LeafCheckResult
+                {
+                    Priority = Priority,
+                    Correct = false,
+                    FeedbackMessage = "There are less parameters than TypeChecks"
+                };
+                subCheckResultsResults.Add(temp);
+                break;
+            }
+
+            for (int x = 0; x < methodParameters.Count(); x++)
+            {
+                var test = methodParameters[x];
+                ICheckResult tempCheck;
+                tempCheck = typecheck.Check(ctx, methodParameters.ElementAt(x));
+
+                switch (tempCheck)
+                {
+                    case LeafCheckResult leafCheckResult:
+                    {
+                        if (leafCheckResult.Correct)
+                        {
+                            subCheckResultsResults.Add(tempCheck);
+                            methodParameters.RemoveAt(x);
+                            noneCorrect = false;
+                            break;
+                        }
+                        continue;
+                    }
+                    case NodeCheckResult nodeCheckResult:
```

Comment:
Currently the check for a nodecheckresult fails when one of its leafs is incorrect. It should be the opposite. It should succeed when at least 1 is correct, and thus fail when all are incorrect.